src/main/java/graphql/annotations/GraphQLAnnotations.java

```diff
@@ -343,11 +314,40 @@ public boolean equals(Object obj) {
         for (Class<?> iface : object.getInterfaces()) {
             if (iface.getAnnotation(GraphQLTypeResolver.class) != null) {
                 builder.withInterface((GraphQLInterfaceType) getInterface(iface));
+                builder.fields(getExtensionFields(iface));
             }
         }
+
+        builder.fields(getExtensionFields(object));
+
         return builder;
     }
 
+    private List<GraphQLFieldDefinition> getExtensionFields(Class<?> object) {
```

Comment:
What happens if I try to change the type of the field? i.e I have
```
Class A {
    String name;
}
```

```
@GraphQLTypeExtension(A.class)
Class B {
    int id;
    int name;
}
```

What happens? Will I get an error, or will name be an int?

Reply:
Right now, it seems to crash. I think we should send a warn/error log and ignore the field.